metric_learn/lmnn.py

```diff
@@ -48,6 +48,9 @@ def _process_inputs(self, X, labels):
     self.X = X
     self.L = np.eye(X.shape[1])
     required_k = np.bincount(self.label_inds).min()
+    if self.params['debug']:
+        # Readying the stats to store the objective function
+        self.stats = np.zeros(self.params['max_iter']-1)
```

Comment:
One potential issue is that `max_iter` may be set to some really large value, which would cause the `self.stats` array to eat way more memory than it needs. It may be better to initialize the stats to an empty list, then coerce the list to a numpy array at the end.